lib/howl/log.moon

```diff
@@ -13,9 +13,29 @@ config.define
   default: 1000
   type_of: 'number'
 
+signal.register 'log-entry-appended',
+  description: 'Called when a new entry is appended to the log'
+  parameters:
+    essentials: 'The log message essentials'
+    level: 'The log level (one of info, warning, error, traceback)'
+    message: 'The log message'
+
+signal.register 'log-trimmed',
+  description: 'Called when the log is trimmed to the given size'
+  parameters:
+    size: 'The new number of entries in the log'
+
 log = {}
 setfenv 1, log
 
+safe_emit = (name, options) ->
+  parent = _G.debug.getinfo 4
+  if parent and parent.name == 'safe_emit' and parent.source\match 'signal.moon$'
```

Comment:
This seems cumbersome and frail - I suggest just using a local state variable defined above instead, e.g. `emitting`.